partiql-parser/src/partiql.pest

```diff
@@ -4,264 +4,549 @@ WHITESPACE = _{ " " | "\t" | "\x0B" | "\x0C" | "\r" | "\n" }
+    | Dec_
+    | Decimal_
+    | Declare_
+    | Default_
+    | Deferrable_
+    | Deferred_
+    | Delete_
+    | Desc_
+    | Describe_
+    | Descriptor_
+    | Diagnostics_
+    | Disconnect_
+    | Distinct_
+    | Domain_
+    | Double_
+    | Drop_
+    | Else_
+    | End_
+    | EndExec_
+    | Escape_
+    | Except_
+    | Exception_
+    | Exec_
+    | Execute_
+    | Exists_
+    | External_
+    | Extract_
+    | DateAdd_
+    | DateDiff_
+    | False_
+    | Fetch_
+    | First_
+    | Float_
+    | For_
+    | Foreign_
+    | Found_
+    | From_
+    | Full_
+    | Get_
+    | Global_
+    | Go_
+    | Goto_
+    | Grant_
+    | Group_
+    | Having_
+    | Identity_
+    | Immediate_
+    | In_
+    | Indicator_
+    | Initially_
+    | Inner_
+    | Input_
+    | Insensitive_
+    | Insert_
+    | Int_
+    | Integer_
+    | Intersect_
+    | Interval_
+    | Into_
+    | Is_
+    | Isolation_
+    | Join_
+    | Key_
+    | Language_
+    | Last_
+    | Left_
+    | Level_
+    | Like_
+    | Local_
+    | Lower_
+    | Match_
+    | Max_
+    | Min_
+    | Module_
+    | Names_
+    | National_
+    | Natural_
+    | Nchar_
+    | Next_
+    | No_
+    | Not_
+    | Null_
+    | Nullif_
+    | Numeric_
+    | OctetLength_
+    | Of_
+    | On_
+    | Only_
+    | Open_
+    | Option_
+    | Or_
+    | Order_
+    | Outer_
+    | Output_
+    | Overlaps_
+    | Pad_
+    | Partial_
+    | Position_
+    | Precision_
+    | Prepare_
+    | Preserve_
+    | Primary_
+    | Prior_
+    | Privileges_
+    | Procedure_
+    | Public_
+    | Read_
+    | Real_
+    | References_
+    | Relative_
+    | Restrict_
+    | Revoke_
+    | Right_
+    | Rollback_
+    | Rows_
+    | Schema_
+    | Scroll_
+    | Section_
+    | Select_
+    | Session_
+    | SessionUser_
+    | Set_
+    | Size_
+    | Smallint_
+    | Some_
+    | Space_
+    | Sql_
+    | Sqlcode_
+    | Sqlerror_
+    | Sqlstate_
+    | Substring_
+    | Sum_
+    | SystemUser_
+    | Table_
+    | Temporary_
+    | Then_
+    | Time_
+    | Timestamp_
+    | To_
+    | Transaction_
+    | Translate_
+    | Translation_
+    | Trim_
+    | True_
+    | Union_
+    | Unique_
+    | Unknown_
+    | Update_
+    | Upper_
+    | Usage_
+    | User_
+    | Using_
+    | Value_
+    | Values_
+    | Varchar_
+    | Varying_
+    | View_
+    | When_
+    | Whenever_
+    | Where_
+    | With_
+    | Work_
+    | Write_
+    | Zone_ 
+}
+
+PartiqlKeyword = _{
+      Missing_
+    | Pivot_
+    | Unpivot_
+    | Limit_
+    | Tuple_
+    | Remove_
+    | Index_
+    | Conflict_
+    | Do_
+    | Nothing_
+    | Returning_
+    | Modified_
+    | All_
+    | New_
+    | Old_
+    | Let_
+    | Bool_
+    | Boolean_
+    | String_
+    | Symbol_
+    | Clob_
+    | Blob_
+    | Struct_
+    | List_
+    | Sexp_
+    | Bag_
+}
+
+//
+// Individual Keyword Definitions
+//
+
+Absolute_ = { ^"absolute" }
+Action_ = { ^"action" }
+Add_ = { ^"add" }
+All_ = { ^"all" }
```

Comment:
This just breaks up every keyword into a rule with an easy to refer to name that should not collide with any real rules.